hpolib/benchmarks/rl/learna.py

```diff
@@ -0,0 +1,362 @@
+    
+    @AbstractBenchmark._configuration_as_dict
+    def objective_function(self, configuration: Dict, cutoff_agent_per_sequence: Union[int, float, None] = 600, **kwargs) \
+            -> Dict:
+
+        config, network_config, agent_config, env_config = self._setup(configuration)
+
+        start_time = time()
+        validation_info = self._validate(evaluation_timeout=cutoff_agent_per_sequence,
+                                         restore_path=None,
+                                         stop_learning=False,
+                                         restart_timeout=config["restart_timeout"],
+                                         network_config=network_config,
+                                         agent_config=agent_config,
+                                         env_config=env_config)
+        cost = time() - start_time
+
+        return {'function_value': validation_info["sum_of_min_distances"],
+                'cost': cost,
+                'num_solved': validation_info["num_solved"],
+                'sum_of_first_distances': validation_info["sum_of_first_distances"],
+                }
+
+    @AbstractBenchmark._configuration_as_dict
+    def objective_function_test(self, configuration: Dict, cutoff_agent_per_sequence: Union[int, float, None] = 600,
+                                **kwargs) -> Dict:
+        return self.objective_function(configuration, cutoff_agent_per_sequence=cutoff_agent_per_sequence, **kwargs)
+
+
+class MetaLearna(BaseLearna):
+    def __init__(self, data_path: Union[str, Path]):
+        super(MetaLearna, self).__init__(data_path)
+        self.config = hpolib.config.config_file
+
+    @AbstractBenchmark._configuration_as_dict
+    def objective_function(self, configuration: Dict, cutoff_agent_per_sequence: Union[int, float, None] = 3600, **kwargs)\
+            -> Dict:
+
+        tmp_dir = Path(tempfile.mkdtemp(dir=self.config.cache_dir))
+
+        config, network_config, agent_config, env_config = self._setup(configuration)
+        start_time = time()
+        cost = 10 ** 10
+        try:
+            train_info = self._train(budget=cutoff_agent_per_sequence,
+                                     tmp_dir=tmp_dir,
+                                     network_config=network_config,
+                                     agent_config=agent_config,
+                                     env_config=env_config)
+
+            validation_info = self._validate(evaluation_timeout=60,
+                                             restore_path=tmp_dir,
+                                             stop_learning=True,
+                                             restart_timeout=config["restart_timeout"],
+                                             network_config=network_config,
+                                             agent_config=agent_config,
+                                             env_config=env_config)
+
+            cost = time() - start_time
+        except Exception as e:
```

Comment:
What kind of exceptions can happen here?
Are there some specific exceptions we want to except and handle differently?

Reply:
This is to catch all possible errors which can occur calling the learna package. 

I cannot say what error may happen here. That's why I've used the broad exception here. But if something happens the temporary directory should be removed.

Reply:
But should be fixed with using try: ... finally: ...

Reply:
Okay, I was just wondering whether there are any expected error (we might want to ignore) because such a broad exception would just catch anything (also error messages that would be helpful to see). But this is fine then